src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp

```diff
@@ -15,214 +15,294 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "ScriptedGossip.h"
 #include "ScriptMgr.h"
-#include "CreatureTextMgr.h"
 #include "GameObject.h"
 #include "MotionMaster.h"
 #include "ObjectAccessor.h"
 #include "Player.h"
 #include "ScriptedEscortAI.h"
-#include "SpellInfo.h"
 #include "SpellScript.h"
 
-/*######
-## npc_koltira_deathweaver
-######*/
+enum BloodyBreakoutTexts
+{
+    SAY_KOLTIRA_0 = 0,
+    SAY_KOLTIRA_1 = 1,
+    SAY_KOLTIRA_2 = 2,
+    SAY_KOLTIRA_3 = 3,
+    SAY_KOLTIRA_4 = 4,
+    SAY_KOLTIRA_5 = 5,
+    SAY_KOLTIRA_6 = 6,
+    SAY_KOLTIRA_7 = 7,
+    SAY_KOLTIRA_8 = 8,
+    SAY_KOLTIRA_9 = 9,
+    SAY_KOLTIRA_10 = 10,
+
+    SAY_VALROTH_0 = 0,
+    SAY_VALROTH_1 = 1,
+    SAY_VALROTH_2 = 2,
+    SAY_VALROTH_3 = 3,
+
+    TEXT_ID_EVENT = 13425
+};
 
-enum Koltira
+enum BloodyBreakoutEvents
 {
-    SAY_BREAKOUT1                   = 0,
-    SAY_BREAKOUT2                   = 1,
-    SAY_BREAKOUT3                   = 2,
-    SAY_BREAKOUT4                   = 3,
-    SAY_BREAKOUT5                   = 4,
-    SAY_BREAKOUT6                   = 5,
-    SAY_BREAKOUT7                   = 6,
-    SAY_BREAKOUT8                   = 7,
-    SAY_BREAKOUT9                   = 8,
-    SAY_BREAKOUT10                  = 9,
-
-    SPELL_KOLTIRA_TRANSFORM         = 52899,
-    SPELL_ANTI_MAGIC_ZONE           = 52894,
-
-    QUEST_BREAKOUT                  = 12727,
-
-    NPC_CRIMSON_ACOLYTE             = 29007,
-    NPC_HIGH_INQUISITOR_VALROTH     = 29001,
-
-    //not sure about this id
-    //NPC_DEATH_KNIGHT_MOUNT        = 29201,
-    MODEL_DEATH_KNIGHT_MOUNT        = 25278
+    EVENT_INTRO_0           = 1,
+    EVENT_INTRO_1           = 2,
+    EVENT_INTRO_2           = 3,
+    EVENT_INTRO_3           = 4,
+    EVENT_INTRO_4           = 5,
+    EVENT_INTRO_5           = 6,
+    EVENT_INTRO_6           = 7,
+
+    EVENT_SPAWN_WAVE_1      = 8,
+    EVENT_SPAWN_WAVE_2      = 9,
+    EVENT_SPAWN_WAVE_3      = 10,
+    EVENT_SPAWN_VALROTH     = 11,
+
+    EVENT_KOLTIRA_ADVICE    = 12,
+    EVENT_OUTRO_1           = 13,
+    EVENT_OUTRO_2           = 14,
+    EVENT_OUTRO_3           = 15,
+    EVENT_OUTRO_4           = 16
 };
 
-class npc_koltira_deathweaver : public CreatureScript
+enum BloodyBreakout
 {
-public:
-    npc_koltira_deathweaver() : CreatureScript("npc_koltira_deathweaver") { }
+    POINT_ID_1                  = 1,
+    POINT_ID_2                  = 2,
+    POINT_ID_6                  = 6,
+    POINT_ID_10                 = 10,
+
+    SUMMON_ACOLYTES_0           = 0,
+    SUMMON_ACOLYTES_1           = 1,
+    SUMMON_ACOLYTES_2           = 2,
+    SUMMON_VALROTH              = 3,
+
+    QUEST_BREAKOUT              = 12727,
+
+    NPC_FAKE_VALROTH            = 29011,
+    NPC_VALROTH                 = 29001,
+    NPC_ACOLYTE                 = 29007,
+    NPC_KOLTIRA                 = 28912,
+    NPC_KOLTIRA_MOUNT           = 25445,
```

Comment:
25445=Nerub'ar Corpse Harvester (Borean Tundra).
Maybe use Koltira's own mount, https://www.wowhead.com/npc=26585/bloodmist instead?

Reply:
25445 is model not the npcID

Reply:
Oh. Of course. Thanks. Feel free to mark this conversation as Resolved.